Source/WebCore/css/CSSSelectorList.h

```diff
@@ -68,6 +68,7 @@ class CSSSelectorList {
     unsigned listSize() const;
 
     CSSSelectorList& operator=(CSSSelectorList&&) = default;
+    static CSSSelectorList resolveNestingParent(const CSSSelectorList& nestedSelectorList, const CSSSelectorList* parentResolvedSelectorList);
```

Comment:
Why can't this stay in `CSSSelectorParser`?

Reply:
It's not really related to the selector parser, it makes more sense to have it in `CSSSelectorList`.

Reply:
Neither `CSSSelectorList` nor `CSSSelector` should be in the business of building themselves. I'd like a factoring where all building and mutation code is separate. It is true that`CSSSelectorParser` may not be a great place for this (we might want `CSSSelectorBuilder` or something) but I don't see why this patch needs to move the function.

Reply:
The refactoring has been moved out of this PR